mne/utils.py

```diff
@@ -1912,3 +1912,76 @@ def compute_corr(x, y):
     # transpose / broadcasting else Y is correct
     y_sd = Y.std(0, ddof=1)[:, None if X.shape == Y.shape else Ellipsis]
     return (fast_dot(X.T, Y) / float(len(X) - 1)) / (x_sd * y_sd)
+
+
+def grand_average(all_inst, interpolate_bads=True, drop_bads=True):
+    """Make grand average of a list evoked or AverageTFR data
+
+    For evoked data, the function interpolates bad channels based on
+    `interpolate_bads` parameter. If `interpolate_bads` is True, the grand
+    average file will contain good channels and the bad channels interpolated
+    from the good MEG/EEG channels.
+    For AverageTFR data, the function takes the subset of channels not marked
+    as bad in any of the instances.
+
+    The grand_average.nave attribute will be equal to the number
+    of evoked datasets used to calculate the grand average.
+
+    Note: Grand average evoked should not be used for source localization.
+
+    Parameters
+    ----------
+    all_inst : list of Evoked or AverageTFR data
+        The evoked datasets.
+    interpolate_bads : bool
+        If True, bad MEG and EEG channels are interpolated. Ignored for
+        AverageTFR.
```

Comment:
what if you have power on planar gradiometers and you interpolate? I suspect it's going to end up badly.

basically you should not interpolate power or phase on MEG gradiometers. You can interpolate the norms
as if it was axial magnetometers (not super clean but should work).

please check it up.